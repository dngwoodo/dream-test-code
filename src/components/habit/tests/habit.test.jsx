import React from 'react';

import { render, screen } from "@testing-library/react";

import userEvent from '@testing-library/user-event';

import renderer from 'react-test-renderer';

import Habit from "../Habit";

describe('Habit', () => {
  const HABIT = { id: 1, name: 'Reading', count: 0 };
  let handleIncrement;
  let handleDecrement;
  let handleDelete;

  beforeEach(() => {
    handleIncrement = jest.fn();
    handleDecrement = jest.fn();
    handleDelete = jest.fn();
  })

  it('renders', () => {
    const component = renderer.create(
      <Habit 
        habit={HABIT}
        onIncrement={handleIncrement}
        onDecrement={handleDecrement}
        onDelete={handleDelete}
      />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('Button Click', () => {
    beforeEach(() => {
      render(
        <Habit 
           habit={HABIT}
           onIncrement={handleIncrement}
           onDecrement={handleDecrement}
           onDelete={handleDelete}
         />
       )
    })

    it('listens increment button click event', () => {
      expect(handleIncrement).toBeCalledTimes(0);

      userEvent.click(screen.getByTitle('increment'));

      expect(handleIncrement).toBeCalledTimes(1);
      expect(handleIncrement).toBeCalledWith(HABIT);
    })

    it('listens decrement button click event', () => {
      expect(handleDecrement).toBeCalledTimes(0);

      userEvent.click(screen.getByTitle('decrement'));

      expect(handleDecrement).toBeCalledTimes(1);
      expect(handleDecrement).toBeCalledWith(HABIT);
    })

    it('listens delete button click event', () => {
      expect(handleDelete).toBeCalledTimes(0);

      userEvent.click(screen.getByTitle('delete'));

      expect(handleDelete).toBeCalledTimes(1);
      expect(handleDelete).toBeCalledWith(HABIT);
    })
  })
});