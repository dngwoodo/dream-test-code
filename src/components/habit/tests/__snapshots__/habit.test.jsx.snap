// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habit renders 1`] = `
<li
  className="habit"
>
  <span
    className="habit-name"
  >
    Reading
  </span>
  <span
    className="habit-count"
  >
    0
  </span>
  <button
    className="habit-button habit-increase"
    onClick={[Function]}
    title="increment"
  >
    <i
      className="fas fa-plus-square"
    />
  </button>
  <button
    className="habit-button habit-decrease"
    onClick={[Function]}
    title="decrement"
  >
    <i
      className="fas fa-minus-square"
    />
  </button>
  <button
    className="habit-button habit-delete"
    onClick={[Function]}
    title="delete"
  >
    <i
      className="fas fa-trash"
    />
  </button>
</li>
`;
